### PR TITLE
Added collaboration option on quick pick

### PR DESCRIPTION
**Pull request by ritikramuka**
image
image

### CODE REVIEW

File: src/web/client/webViews/QuickPickProvider.ts [hunk 1]
```diff
@@ -44,11 +51,51 @@ export class QuickPickProvider {
...
+        if (selectedUser) {
+            this.handleSelectedOption(selectedUser);
+        }
+    }
+
+    private handleSelectedOption(selectedOption: IQuickPickItem) {
+        this.collaborationQuickPick(selectedOption);
+    }
+
+    public async collaborationQuickPick(selectedOption: IQuickPickItem) {
```
Comment (vedg) [inline src/web/client/webViews/QuickPickProvider.ts:70]: nit: rename this CollaborationQuickPick  to handleCollaborationQuickPick

File: src/web/client/webViews/QuickPickProvider.ts [hunk 2]
```diff
@@ -44,11 +51,51 @@ export class QuickPickProvider {
...
+
+        if (collaborationOptionsSelected) {
+            if (collaborationOptionsSelected.label === Constants.START_TEAMS_CHAT) {
+                if (selectedOption.id) {
+                    WebExtensionContext.openTeamsChat(selectedOption.id);
+                }
+            } else if (collaborationOptionsSelected.label === Constants.SEND_AN_EMAIL) {
+                if (selectedOption.id) {
+                    WebExtensionContext.openMail(selectedOption.id);
+                }
```
Comment (vedg) [inline src/web/client/webViews/QuickPickProvider.ts:97]: you can have separate methods to handle these two separate logic of opening mail and teams chat

File: src/web/client/webViews/QuickPickProvider.ts [hunk 3]
```diff
@@ -44,11 +51,51 @@ export class QuickPickProvider {
...
+
+    private handleSelectedOption(selectedOption: IQuickPickItem) {
+        this.collaborationQuickPick(selectedOption);
+    }
+
+    public async collaborationQuickPick(selectedOption: IQuickPickItem) {
+        const collaborationOptions: IQuickPickItem[] = [
+            {
+                label: Constants.START_TEAMS_CHAT,
+                iconPath: {
```
Comment (vedg) [inline src/web/client/webViews/QuickPickProvider.ts:74]: use relative paths using "vscode.Uri.file"

File: src/web/client/webViews/QuickPickProvider.ts [hunk 4]
```diff
@@ -44,11 +51,51 @@ export class QuickPickProvider {
...
+            },
+        ];
+
+        const collaborationOptionsSelected = await vscode.window.showQuickPick(collaborationOptions, {
+            title: `CONTACT ${selectedOption.label.toUpperCase()}`,
+        });
+
+        if (collaborationOptionsSelected) {
+            if (collaborationOptionsSelected.label === Constants.START_TEAMS_CHAT) {
+                if (selectedOption.id) {
```
Comment (vedg) [inline src/web/client/webViews/QuickPickProvider.ts:91]: handle case for selectedOption.id to be undefined, add error handling

File: src/web/client/WebExtensionContext.ts [hunk 5]
```diff
@@ -783,6 +789,18 @@ class WebExtensionContext implements IWebExtensionContext {
...
     }
+
+    public async openTeamsChat(userId: string): Promise<void> {
+        const mail = await this.graphClientService.getUserEmail(userId);
+        const teamsChatLink = getTeamChatURL(mail);
+        vscode.env.openExternal(vscode.Uri.parse(teamsChatLink.href));
+    }
+
+    public async openMail(userId: string): Promise<void> {
+        const mail = await this.graphClientService.getUserEmail(userId);
```
Comment (vedg) [inline src/web/client/WebExtensionContext.ts:800]: handle case when mail is not retrieved - this.graphClientService.getUserEmail(userId)

Comment (vedg) [inline src/web/client/WebExtensionContext.ts:800]: similar logic, code duplication can be avoided here

Comment (ritikramuka) [reply src/web/client/WebExtensionContext.ts:800]: addressed

File: src/web/client/webViews/QuickPickProvider.ts (hunk 4 above)
Comment (ritikramuka) [reply src/web/client/webViews/QuickPickProvider.ts:91]: addressed!

File: src/web/client/webViews/QuickPickProvider.ts (hunk 2 above)
Comment (ritikramuka) [reply src/web/client/webViews/QuickPickProvider.ts:101]: can you explain this bit more! didn't get what can be done here

File: src/web/client/webViews/QuickPickProvider.ts [hunk 6]
```diff
@@ -44,11 +51,55 @@ export class QuickPickProvider {
...
             this.items = [{
                 label: Constants.WEB_EXTENSION_QUICK_PICK_DEFAULT_STRING,
+                id: "",
             }];
         }
     }
 
-    public showQuickPick() {
-        vscode.window.showQuickPick(this.items);
+    public async showQuickPick() {
```
Comment (amitjoshi438) [inline src/web/client/webViews/QuickPickProvider.ts:59]: nit: can we rename this to selectUserWithQuickPick()? showQuickPick() is native method of VSCode

File: src/web/client/webViews/QuickPickProvider.ts (hunk 2 above)
Comment (vedg) [reply src/web/client/webViews/QuickPickProvider.ts:101]: this is called "Separation of Concern" - separating the logic for handling different collaboration options into distinct methods. this is for better maintainability and modularity.

**Review by amitjoshi438 — APPROVED** [review]

**Review by vedg — APPROVED** [review]
take care of nit comments in next PR. also please check into the build failures

File: src/web/client/webViews/QuickPickProvider.ts (hunk 6 above)
Comment (ritikramuka) [reply src/web/client/webViews/QuickPickProvider.ts:59]: the method was only invoking that showQuickPick method of vscode, thats why gave same name

**Review by tehcrashxor — APPROVED** [review]

**Pull request merged**